[TASK-131] feaure(root-admin): 루트 어드민 festival 작업

hyunwleeee opened the pull request.

hyunwleeee committed:
mv: table-board to concert-board

hyunwleeee committed:
feat: getAllFestival fetch api

hyunwleeee committed:
Merge branch 'chore/54-migrate-to-monorepo' of https://github.com/ToyVallet/swifty-frontend into feature(root-admin)/131-festival-page

hyunwleeee committed:
feat: concert 화면, API 연동

hyunwleeee committed:
feat: festival 화면, GET(total, detail) API 연동

hyunwleeee committed:
feat: card 컴포넌트 구현

hyunwleeee committed:
feat: drawer-button 컴포넌트 구현

hyunwleeee committed:
feat: lineup 화면, API 연동 중

hyunwleeee committed:
Merge branch 'chore/54-migrate-to-monorepo' of https://github.com/ToyVallet/swifty-frontend into feature(root-admin)/131-festival-page

hyunwleeee commented on apps/root-admin/src/app/components/festival/form/concert-update-form.tsx:
```diff
@@ -1,80 +1,162 @@
+        endDate: null
+      });
+    }
+  }, [isLock]);
```
initialValues의 startDate, endDate 값들이 DatePicker(antd)에서 이상하게 보여지는 현상이 있어서 disabled가 아닐 때 null로 값을 변환시켰습니다. null로 변환시키지 않도록 찾아볼게요.

kd02109 commented on apps/root-admin/.yarn/install-state.gz:
사용하시는 yarn 버전을 1.22.21로 변경하면, 해당 .yarn 폴더가 생성되지 않을꺼에요!
yarn classic이랑 yarn berry의 동작방식이 다르다는 걸 이번에 알게 되어서 참고 하면 좋을 것 같습니다.!
https://helloinyong.tistory.com/343

kd02109 commented on apps/root-admin/src/app/(dashboard)/festivals/[id]/data.ts:
해당 api 요청을 page에서 하시지 않고 별도의 data 파일로 뺀 의도가 있을까요?

kd02109 commented on apps/root-admin/src/app/(dashboard)/festivals/[id]/page.tsx:
```diff
@@ -1,22 +1,28 @@
-    await getMockFestivalDetail(id);
+  const {
+    adminFestivalInfoResponse,
+    adminConcertInfoResponses
```
결국 저희가 사용하는 데이터는 각 페스티벌 정보, 콘서트 정보이고 Response는 API와 연관된 용어라고 생각합니다. 그래서 Response는 해당 데이터의 역할과 연관이 없다고 생각해서 아래와 같이 표현하는 건 어떨까요?
```suggestion
    adminFestivalInfo,
    adminConcertInfo
```

kd02109 commented on apps/root-admin/src/app/components/festival/concert-board/index.tsx:
```diff
@@ -0,0 +1,97 @@
+}: { festivalSubId: string; concertsInfo: ConcertsResponse[]; }) {
+  const defaultPanes = concertsInfo.map(({ subId, name }) => {
+    return { label: name, key: subId };
+  });
```
`defaultPanes`라는 이름의 변수가 어떤 것을 의미하는지 잘 이해가 되지 않습니다. 콘서트의 이름과 아이디를 새롭게 배열로 만드는 것 같은데, 어 떤 역할을 하는지 더 명시적인 이름이 있으면 좋을 것 같아요!

kd02109 commented on apps/root-admin/src/app/components/festival/concert-board/index.tsx:
```diff
@@ -0,0 +1,97 @@
+  });
+  const [activeKey, setActiveKey] = useState(defaultPanes[0]?.key);
+  const [items, setItems] = useState(defaultPanes);
+  const activeConcertInfo = concertsInfo.filter(({ subId }) => subId === activeKey)[0];
```
active 콘서트를 정하는 기준이 어떤 것인지 궁금합니다. 그냥 해당 콘서트가 존재하면 모두 active concert인가요?

kd02109 commented on apps/root-admin/src/app/components/festival/concert-board/index.tsx:
```diff
@@ -0,0 +1,97 @@
+  }));
+
+  const removeLineup = async (subId: string) => {
+    if (confirm("정말로 삭제하시겠습니까?"))
```
`confirm`을 사용하는 방법도 좋지만, antd를 활용하고 있으니 antd의 Popconfirm 을 활용하는 건 어떨까요?

kd02109 commented on apps/root-admin/src/app/components/festival/drawer-button/index.tsx:
```diff
@@ -34,7 +38,7 @@ export default function DrawerButton({
-        className={clsx({ [styles.createButton]: variant.includes('create') }, className)}
+        className={clsx({ [styles.createButton as string]: variant.includes('create') }, className)}
         type={variant.includes('create') ? 'primary' : 'default'}
         style={variant.includes('create') ? { height: '3rem', borderRadius: '1.5rem', fontSize: '14px', fontWeight: '700' } : {}}
```
className과 style로 각각 style을 관리하기 보다는, css module과 clsx를 함께 써서 className으로 함께 관리하는 건 어떨까요? 지금 방식은 css 관리 포인트가 두 군데로 분산되어 있어서 하나로 통일하면 좋을 듯 합니다.

kd02109 commented on apps/root-admin/src/app/hooks/festival/useConcert.tsx:
useConcert 보다는 해당 API가 concert  생성 삭제 수정과 연관있으니 좀더 명확하게 `useConcertCRUD`는 어떨까요?

kd02109 reviewed (commented):
수고하셨습니다 !!

hyunwleeee committed:
chore: remove moment, add dayjs

hyunwleeee committed:
fix: datePicker 날짜 오류 수정

hyunwleeee committed:
mv: card to panel

hyunwleeee committed:
feat: open, hidden concert api (optimistic 업데이트)

kd02109 committed:
feat: University Table 생성

kd02109 committed:
feat: 대학교 정보 수정 작성

kd02109 committed:
feat: 대학 생성, 대학 로고 수정

kd02109 committed:
fix: 대학 생성 버튼 페이지 수정

hyunwleeee committed:
feat: concert api 요청

hyunwleeee committed:
feat: lineup api 요청

kd02109 committed:
feat: 대학교 검색 API 연결

kd02109 committed:
fix: 대학 테이블 column 로고 주소 정보 추가

kd02109 committed:
feat: 배럴 파일 도입

kd02109 committed:
fix: 대학 생성 logo 로직 수정

hyunwleeee committed:
Merge branch 'dev' of https://github.com/ToyVallet/swifty-frontend into feature(root-admin)/131-festival-page

hyunwleeee committed:
feat: 임시 코드 in festival

1. 급한대로 multipart-form 전송하면서 custom-fetch 함수 헤더의 application/json을
   덮어쓰기를 하지 못해 주석처리하고 사용했습니다. multipart-form에
   관해서 migration 해주시면 좋을 것 같습니다.
2. 낙관적 없데이트 결국 마무리 못 지었습니다. 콘서트 open, hidden처리
   미완성
3. 라인업 open, hidden 처리 아직 못했습니다. (백엔드 api TASK-194
   기다리는 중)
4. 라인업 update 처리 아직 못했습니다. (백엔드 api에서 이전 이미지가 null일 경우 수정이 안되는 부분 수정
   기다리는 중)

kd02109 committed:
Merge branch 'dev' into feature(root-admin)/150-university

kd02109 committed:
feat: University detail page breadcrumb

kd02109 committed:
feat: 대학 상세 페이지 호스트 유저 조회 기능

kd02109 committed:
feat: 대학 상세 페이지 축제 보기

kd02109 committed:
feat: 대학 축제 생성 기능 추가

kd02109 committed:
refactor: delte button 컴포넌트 구분

kd02109 committed:
Merge branch 'feature(root-admin)/199-detail-university-festivals' into feature(root-admin)/131-festival-page

kd02109 committed:
refactor: 불필요한 코드 제거

kd02109 committed:
fix: 콘서트 업데이트 수정

kd02109 committed:
fix: 콘서트 생성 로직 수정

kd02109 committed:
fix: 라인업 생성 경로 수정

kd02109 committed:
fix: 라인업 수정 로직 수정

kd02109 committed:
feat: 라인업 삭제

kd02109 committed:
feat: 축제, 콘서트, 라인업 상태 변경 로직 추가

kd02109 committed:
feat: 콘서트 페이지 콘서트 수정, 삭제, 라인업 추가 버튼 생성

kd02109 committed:
feat : Festival 수정 삭제 기능 추가

kd02109 committed:
fix: 대학 축제 수정 이미지 처리 로직 수정

kd02109 committed:
fix: form 데이터 초기화 로직 추가

kd02109 committed:
feat: 콘서트 페이지 페스티벌 description 추가

kd02109 commented:
- [TASK-197] 작업을 머지 하고 이루어 져서 파일변화가 많습니다...
- 현우님 코드에서 살릴 수 있는 코드를 살리고 수정을 진행했습니다. 
- 현재 pr 중에서 가장 마지막에 진행해주시면 될 것 같습니다.! 
- 주요 반영 사항은 축제 CRUD, 콘서트 CRUD, 페스티벌 CRUD 입니다.


https://github.com/ToyVallet/swifty-frontend/assets/57277708/79ae2724-a142-430e-b792-44d738fe0e8a

ho991217 reviewed (approved):
LGTM...
